Softeq.XToolkit.PushNotifications.Droid/DroidPushNotificationParser.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Developed by Softeq Development Corporation
+// http://www.softeq.com
+
+using System.Collections.Generic;
+using Android.OS;
+using Firebase.Messaging;
+
+namespace Softeq.XToolkit.PushNotifications.Droid
+{
+    public class DroidPushNotificationParser : IPushNotificationParser
+    {
+        /// <summary>
+        /// Custom key for title (stored inside RemoteMessage.Data). Can be customized.
+        /// </summary>
+        protected virtual string DataTitleKey => "title";
+
+        /// <summary>
+        /// Custom key for addition body (stored inside RemoteMessage.Data). Can be customized.
+        /// </summary>
+        protected virtual string DataBodyKey => "body";
+
+        /// <summary>
+        /// Custom key for addition data (stored inside RemoteMessage.Data). Can be customized.
+        /// </summary>
+        protected virtual string DataKey => "data";
+
+        public PushNotificationModel Parse(object pushNotificationData)
+        {
+            var pushNotification = new PushNotificationModel();
+
+            if (pushNotificationData is RemoteMessage)
```

Comment:
create different method to parse different type of messages

Reply:
created separate methods